### DISCUSSION TITLE
What is the long-term and short-term goal?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson-python#636] [Q&A] [answered]
What is the long-term and short-term goal of the Meson Python layer?

**Comment by rgommers** [comment, marked as answer]
The first paragraph of the [documentation](https://mesonbuild.com/meson-python/) states the goals/purpose of `meson-python`: _"meson-python implement the Python build system hooks, enabling Python build front-ends such as pip and build to build and install Python packages based on a [Meson](https://mesonbuild.com/) build definition."_

The goals in the short and long term don't differ: simply make it as good an experience as possible to use Meson for building Python packages.

**Reply by dreamer-coding** [reply]
Ok so let's say I do a Python project, this project has a lot of documentation and use this Meson python backend, can I use it to handle generating docs with the Meson Hotdocs module?

**Reply by rgommers** [reply]
No. This has never come up - Hotdoc is an extremely niche documentation generator. Best to use MkDocs, Sphinx, or something else more standard for a Python project.

**Reply by dnicolodi** [reply]
> No.

@rgommers Why not? meson-python does not care about how a target is generated, as long as there is a meson rule that generates it. However, documentation is not usually included in Python wheels. Thus, most likely, meson-python does not even have to know that there is some documentation generated.

@dreamer-coding-555 Is this only an hypothetical scenario or do you have a concrete use case?

**Reply by rgommers** [reply]
> as long as there is a meson rule that generates it

Oh sure you can build anything, but if it doesn't land in `intro-install_plan.json` nor has a location in the wheel that it can be included in, then I'd say it's not supported by (or irrelevant to) `meson-python`.

**Reply by dreamer-coding** [reply]
@dnicolodi I was hoping to have the Meson-UI project generate the docs using HotDocs. I assume since `meson-python` seems to provide a backend for using the Meson build system in the background it might be possible to use HotDocs.

**Reply by eli-schwartz** [reply]
> No. This has never come up - Hotdoc is an extremely niche documentation generator. Best to use MkDocs, Sphinx, or something else more standard for a Python project.

> Oh sure you can build anything, but if it doesn't land in `intro-install_plan.json` nor has a location in the wheel that it can be included in, then I'd say it's not supported by (or irrelevant to) `meson-python`.

@rgommers,

I agree that hotdoc isn't supported by / irrelevant to meson-python.

And by the same logic, mkdocs and sphinx aren't supported by / irrelevant to meson-python...

It's the docs part that poses the issue, not the individual tool used. Python wheels don't have a standard for installing documentation. And even though Unix build systems do have a standard (docdir) most people prefer the authors to run the build system for the docs, then upload the results to a website, rather than also building the docs themselves.

**Reply by dnicolodi** [reply]
> I was hoping to have the Meson-UI project generate the docs using HotDocs. I assume since `meson-python` seems to provide a backend for using the Meson build system in the background it might be possible to use HotDocs.

I don't understand which role meson-python has in what you are trying to build. In particular, I don't understand what the question about the short- and long-term goals of meson-python arose.

> `meson-python` seems to provide a backend for using the Meson build system in the background

What do you mean with "in the background" in this context? As @rgommers explained, meson-python provides the glue to use a Python package build front-end such as `pip` or `build` to construct a Python wheel binary distribution from a project that uses the Meson build system to describe its build process.

**Reply by dreamer-coding** [reply]
See if I can rebase this, since `meson-python` acts as a bridge between Meson and PiPy does it inherit the ability to use HotDocs same way as with Meson build where you just need to get the tools for HotDocs?

**Reply by dnicolodi** [reply]
> since `meson-python` acts as a bridge between Meson and PiPy

This is not correct.

> does it inherit the ability to use HotDocs same way as with Meson build where you just need to get the tools for HotDocs?

meson-python just runs meson, takes the build artifacts, and stuffs them into a wheel. I don't know how to explain this in a different way that may make it more clear. If you want to include the built documentation into the binary distribution of a Python package, you can do that with meson-python. However, that's not what is usually done and I don't see a reason why anyone would want to do that.
